src/ol/source/tilewmssource.js

```diff
@@ -277,6 +283,19 @@ ol.source.TileWMS.prototype.getRequestUrl_ =
 
 
 /**
+ * @param {ol.proj.Projection} projection Projection.
+ * @return {ol.tilegrid.TileGrid} Tile grid.
+ */
+ol.source.TileWMS.prototype.getTileGridForProjection = function(projection) {
```

Comment:
This might be useful in other contexts as well. So probably worth creating `ol.tilegrid.getForProjectionAndMaxZoom` and cache there?